project-base/config/services.yaml

```diff
@@ -123,3 +123,12 @@ services:
     Psr\Http\Message\ServerRequestInterface:
         factory: ['@Symfony\Bridge\PsrHttpMessage\Factory\DiactorosFactory', 'createRequest']
         arguments: ['@=service("request_stack").getMasterRequest()']
+
+    Shopsys\FrameworkBundle\Component\Router\FriendlyUrl\FriendlyUrlRouterFactory:
+        arguments:
+            $friendlyUrlRouterResourceFilepath: '%shopsys.router.friendly_url_router_filepath%'
+            $mainFriendlyUrlSlugCache: '@cache.main_friendly_url_slugs'
```

Comment:
I'm thinking about using Symfony cache pool behavior.
The proper cache is auto-wirable if the argument is named similarly to cache pool name
 e.g. if you name pool `main_friendly_url_slug_cache` in `framework.yaml`  (without `cache.` prefix)

 it can be seen as magic but in the end, it's friendlier to the developers
- you can remove both service definitions (no need to specify the service manually)
- developers don't need to specify the correct cache when extending ( right now it's error-prone if I extend the class and forget about argument)
- it solves the problem with the generator as a service (my first comment)

see 
https://symfony.com/doc/current/cache.html#creating-custom-namespaced-pools

Reply:
Thakns for teaching me a new symfony magic!

Magic was applied.